inline_verifier.go

```diff
@@ -3,31 +3,158 @@ package ghostferry
 import (
 	"bytes"
 	"database/sql"
+	"errors"
 	"fmt"
 	"strconv"
 	"strings"
+	"sync"
+	"time"
 
 	"github.com/golang/snappy"
 	"github.com/sirupsen/logrus"
 )
 
-type InlineVerifier struct {
-	SourceDB *sql.DB
-	TargetDB *sql.DB
+// This struct is very similar to ReverifyStore, but it is more optimized
+// for serialization into JSON.
+//
+// TODO: remove IterativeVerifier and remove this comment.
+type BinlogVerifyStore struct {
+	EmitLogPerRowsAdded uint64
+
+	mutex *sync.Mutex
+	store map[string]map[string]map[uint64]struct{} // db => table => (pk set)
+
+	// The total number of rows added to the reverify store, ever.  Does not
+	// include the rows added in the interrupted run if the present run is a
+	// resuming one. This is only used for emitting metrics.
+	totalRowCount   uint64
+	currentRowCount uint64 // The number of rows in store currently.
```

Comment:
The problem is the currentRowCount is nominally close to 0 all the time, whereas we emit a log every 1000 entries. Previously (i.e. within the IterativeVerifier) this worked because the reverify queue mostly only grew. Right now, it keeps getting emptied every 1 second.

Reply:
That makes sense but I would like to make the purpose of it clearer. This is not much more than a monotonically increasing integer (doesn't have to be serialized or exposed to the user at all), right? 

Alternatively, could we step back here and consider how else we could throttle the logs? Maybe using a timestamp?